pkg/model/resources/nodeup.go

```diff
@@ -76,29 +92,28 @@ download-or-bust() {
 
   while true; do
     for url in "${urls[@]}"; do
-      commands=(
-        "curl -f --compressed -Lo "${file}" --connect-timeout 20 --retry 6 --retry-delay 10"
-        "wget --compression=auto -O "${file}" --connect-timeout=20 --tries=6 --wait=10"
-        "curl -f -Lo "${file}" --connect-timeout 20 --retry 6 --retry-delay 10"
-        "wget -O "${file}" --connect-timeout=20 --tries=6 --wait=10"
-      )
-      for cmd in "${commands[@]}"; do
-        echo "Attempting download with: ${cmd} {url}"
-        if ! (${cmd} "${url}"); then
-          echo "== Download failed with ${cmd} =="
-          continue
-        fi
-        if ! validate-hash "${file}" "${hash}"; then
-          echo "== Hash validation of ${url} failed. Retrying. =="
-          rm -f "${file}"
-        else
-          echo "== Downloaded ${url} (SHA256 = ${hash}) =="
-          return 0
-        fi
-      done
+      local access_token=""
+      if [[ "${GCE_ACCESS_TOKEN_URL:-}" != "" ]]; then
+        # The token always begins with 'ya29' and continues until the next double-quote.
+        # We would normally use jq to be more tolerant of format changes, 
+        # but it's not available in flatcar during startup (depends on docker, which isn't up yet.)
+        access_token=$(curl "$GCE_ACCESS_TOKEN_URL" -H "Metadata-Flavor: Google" | grep -Eo 'ya29\.[^\"]*')
+      fi
+
+      echo "Attempting download of ${url}"
+      if ! try-download-file "${url}" "${file}" "${access_token}" ; then
+        continue
+      fi
+      if ! validate-hash "${file}" "${hash}"; then
+        echo "== Hash validation of ${url} failed. Retrying. =="
+        rm -f "${file}"
+      else
+        echo "== Downloaded ${url} (SHA256 = ${hash}) =="
+        return
+      fi
     done
 
-    echo "All downloads failed; sleeping before retrying"
+    echo "Download failed with all plausible commands; sleeping before retrying"
```

Comment:
By any chance you meant `possible` instead of `plausible`?

Reply:
"possible" reads a lot cleaner, yeah.